src/store/collectibles/collectibles-epics.ts

```diff
@@ -0,0 +1,37 @@
+import { combineEpics } from 'redux-observable';
+import { Observable, of } from 'rxjs';
+import { catchError, map, switchMap } from 'rxjs/operators';
+import { Action } from 'ts-action';
+import { ofType, toPayload } from 'ts-action-operators';
+
+import { fetchCollectibleFloorPrice$ } from '../../apis/objkt';
+import { loadAllCollectiblesDetails$ } from '../../utils/collectibles.utils';
+import { updateCollectibleDetailsAction, loadCollectiblesDetailsActions } from './collectibles-actions';
+
+const loadCollectiblesDetailsEpic = (action$: Observable<Action>) =>
+  action$.pipe(
+    ofType(loadCollectiblesDetailsActions.submit),
+    toPayload(),
+    switchMap(collectiblesSlugs =>
+      loadAllCollectiblesDetails$(collectiblesSlugs).pipe(
+        map(collectiblesDetails => loadCollectiblesDetailsActions.success(collectiblesDetails)),
+        catchError(err => of(loadCollectiblesDetailsActions.fail(err.message)))
+      )
+    )
+  );
+
+const loadCollectibleFloorPriceEpic = (action$: Observable<Action>) =>
+  action$.pipe(
+    ofType(updateCollectibleDetailsAction.submit),
+    toPayload(),
+    switchMap(collectible =>
+      fetchCollectibleFloorPrice$(collectible.address, collectible.id.toString()).pipe(
+        map(listingsActive =>
+          updateCollectibleDetailsAction.success(JSON.stringify({ ...collectible, listingsActive }))
```

Comment:
why we need to pass JSON.stringify ? can we pass just object?

Reply:
I don't know, I wanted to discuss it at the previous meeting but we didn't. I had an error with typescript if I pass an object but not a string